controllers/alerts/alerts.go

```diff
@@ -0,0 +1,236 @@
+func NewPrometheusRuleSpec() *monitoringv1.PrometheusRuleSpec {
+	return &monitoringv1.PrometheusRuleSpec{
+		Groups: []monitoringv1.RuleGroup{{
+			Name: alertRuleGroup,
+			Rules: []monitoringv1.Rule{
+				createOutOfBandUpdateAlertRule(),
+				createUnsafeModificationAlertRule(),
+				createInstallationNotCompletedAlertRule(),
+				createRequestCPUCoresRule(),
+			},
+		}},
+	}
+}
+
+func createOutOfBandUpdateAlertRule() monitoringv1.Rule {
+	return monitoringv1.Rule{
+		Alert: outOfBandUpdateAlert,
+		Expr:  intstr.FromString("sum by(component_name) ((round(increase(kubevirt_hco_out_of_band_modifications_count[10m]))>0 and kubevirt_hco_out_of_band_modifications_count offset 10m) or (kubevirt_hco_out_of_band_modifications_count != 0 unless kubevirt_hco_out_of_band_modifications_count offset 10m))"),
+		Annotations: map[string]string{
+			"description": "Out-of-band modification for {{ $labels.component_name }}.",
+			"summary":     "{{ $value }} out-of-band CR modifications were detected in the last 10 minutes.",
+			"runbook_url": outOfBandUpdateRunbookUrl,
+		},
+		Labels: map[string]string{
+			severityAlertLabelKey:  "warning",
+			partOfAlertLabelKey:    partOfAlertLabelValue,
+			componentAlertLabelKey: componentAlertLabelValue,
+		},
+	}
+}
+
+func createUnsafeModificationAlertRule() monitoringv1.Rule {
+	return monitoringv1.Rule{
+		Alert: unsafeModificationAlert,
+		Expr:  intstr.FromString("sum by(annotation_name) ((kubevirt_hco_unsafe_modification_count)>0)"),
+		Annotations: map[string]string{
+			"description": "unsafe modification for the {{ $labels.annotation_name }} annotation in the HyperConverged resource.",
+			"summary":     "{{ $value }} unsafe modifications were detected in the HyperConverged resource.",
+			"runbook_url": unsafeModificationRunbookUrl,
+		},
+		Labels: map[string]string{
+			severityAlertLabelKey:  "info",
+			partOfAlertLabelKey:    partOfAlertLabelValue,
+			componentAlertLabelKey: componentAlertLabelValue,
+		},
+	}
+}
+
+func createInstallationNotCompletedAlertRule() monitoringv1.Rule {
+	return monitoringv1.Rule{
+		Alert: installationNotCompletedAlert,
+		Expr:  intstr.FromString("kubevirt_hco_hyperconverged_cr_exists == 0"),
+		Annotations: map[string]string{
+			"description": "the installation was not completed; the HyperConverged custom resource is missing. In order to complete the installation of the Hyperconverged Cluster Operator you should create the HyperConverged custom resource.",
+			"summary":     "the installation was not completed; to complete the installation, create a HyperConverged custom resource.",
+			"runbook_url": installationNotCompletedRunbookUrl,
+		},
+		For: "1h",
+		Labels: map[string]string{
+			severityAlertLabelKey:  "info",
+			partOfAlertLabelKey:    partOfAlertLabelValue,
+			componentAlertLabelKey: componentAlertLabelValue,
+		},
+	}
+}
+
+// Recording rules for openshift/cluster-monitoring-operator
+func createRequestCPUCoresRule() monitoringv1.Rule {
```

Comment:
is this needed?

Reply:
yes. it was there before the PR. Just wrapped it in a function